pkg/controller/gitopsservice/gitopsservice_controller.go

```diff
@@ -287,168 +307,191 @@ func (r *ReconcileGitopsService) Reconcile(request reconcile.Request) (reconcile
-		return reconcile.Result{}, err
-	}
+	// Create backend Service
+	{
+		serviceRef := newBackendService(gitopsserviceNamespacedName)
+		// Set GitopsService instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, serviceRef, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
 
-	// Check if this Service already exists
-	existingServiceRef := &corev1.Service{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: serviceRef.Name, Namespace: serviceRef.Namespace}, existingServiceRef)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new Service", "Namespace", serviceRef.Namespace, "Name", serviceRef.Name)
-			err = r.client.Create(context.TODO(), serviceRef)
-			if err != nil {
+		// Check if this Service already exists
+		existingServiceRef := &corev1.Service{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: serviceRef.Name, Namespace: serviceRef.Namespace},
+			existingServiceRef); err != nil {
+
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new Service", "Namespace", serviceRef.Namespace, "Name", serviceRef.Name)
+				err = r.client.Create(context.TODO(), serviceRef)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			} else {
 				return reconcile.Result{}, err
 			}
-		} else {
-			return reconcile.Result{}, err
 		}
 	}
 
-	routeRef := newBackendRoute(serviceNamespacedName)
-	// Set GitopsService instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, routeRef, r.scheme); err != nil {
-		return reconcile.Result{}, err
-	}
+	// Create backend Route
+	{
+		routeRef := newBackendRoute(gitopsserviceNamespacedName)
+		// Set GitopsService instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, routeRef, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
 
-	existingRoute := &routev1.Route{}
+		existingRoute := &routev1.Route{}
 
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: routeRef.Name, Namespace: routeRef.Namespace}, existingRoute)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new Route", "Namespace", routeRef.Namespace, "Name", routeRef.Name)
-			err = r.client.Create(context.TODO(), routeRef)
-			if err != nil {
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: routeRef.Name, Namespace: routeRef.Namespace},
+			existingRoute); err != nil {
+
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new Route", "Namespace", routeRef.Namespace, "Name", routeRef.Name)
+				err = r.client.Create(context.TODO(), routeRef)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			} else {
 				return reconcile.Result{}, err
 			}
-		} else {
-			return reconcile.Result{}, err
 		}
 	}
 
-	return r.reconcileCLIServer(instance, request)
+	return reconcile.Result{}, nil
```

Comment:
FYI the call to reconcileCLIServer is still here, it's just been moved to the parent.